uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrix.php

```diff
@@ -49,7 +49,7 @@ public function canTransitionTo($fromStatus, $toStatus, $leaveRequestContactID)
     }
 
     if (empty($statusMatrix)) {
-      throw new RuntimeException('You are not authorized to update this leave request');
+      return false;
     }
```

Comment:
One small possible improvement here: Extract the logic to get the matrix for the current user (lines 40 to 49) to a separate method.